tests/python/pants_test/cache/test_s3_artifact_cache.py

```diff
@@ -0,0 +1,198 @@
+# coding=utf-8
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+from contextlib import contextmanager
+
+import boto3
+import pytest
```

Comment:
It was an intentional choice, although not necessarily the right one. Pytest certainly encourages you not to, and there are some features unavailable to class-based tests.  Nonetheless, fixtures are confusing, and xtest-style tests are well understood by most, so...